kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/logs/TestLogCollector.java

```diff
@@ -0,0 +1,232 @@
+                                    }
+                                })
+                                .sorted()
+                                .toList()
+                                .get(filesInLogsDir.length - 1))
+                        + 1;
+            }
+        }
+
+        return rootPathToLogsForTestCase.resolve(String.valueOf(index));
+    }
+
+    /**
+     * Method for building the full path to logs for specified test-class and test-case.
+     *
+     * @param testClass     name of the test-class
+     * @param testCase      name of the test-case
+     *
+     * @return full path to the logs for test-class and test-case, together with index
+     */
+    private Path buildFullPathToLogs(String testClass, String testCase) {
+        Path rootPathToLogsForTestCase = Path.of(Environment.CLUSTER_DUMP_DIR, CURRENT_DATE, testClass);
+
+        if (testCase != null) {
+            rootPathToLogsForTestCase = rootPathToLogsForTestCase.resolve(testCase);
+        }
+
+        return checkPathAndReturnFullRootPathWithIndexFolder(rootPathToLogsForTestCase);
+    }
+
+    /**
+     * Method that encapsulates {@link #collectLogs(String, String)}, taking the test-class and test-case names from
+     * {@link ResourceManager#getTestContext()}
+     */
+    public void collectLogs() {
+        collectLogs(
+                ResourceManager.getTestContext().getRequiredTestClass().getName(),
+                ResourceManager.getTestContext().getRequiredTestMethod().getName());
+    }
+
+    /**
+     * Method that encapsulates {@link #collectLogs(String, String)}, where test-class is passed as a parameter and the
+     * test-case name is `null` -> that's used when the test fail in `@BeforeAll` or `@AfterAll` phases.
+     *
+     * @param testClass     name of the test-class, for which the logs should be collected
+     */
+    public void collectLogs(String testClass) {
+        collectLogs(testClass, null);
+    }
+
+    /**
+     * Method that uses {@link LogCollector#collectFromNamespaces(String...)} method for collecting logs from Namespaces
+     * for the particular combination of test-class and test-case.
+     *
+     * @param testClass     name of the test-class, for which the logs should be collected
+     * @param testCase      name of the test-case, for which the logs should be collected
+     */
+    public void collectLogs(String testClass, String testCase) {
+        Path rootPathToLogsForTestCase = buildFullPathToLogs(testClass, testCase);
+
+        final LogCollector testCaseCollector = new LogCollectorBuilder(logCollector)
+                .withRootFolderPath(rootPathToLogsForTestCase.toString())
+                .build();
+
+        List<String> namespaces = NamespaceUtils.getListOfNamespacesForTestClassAndTestCase(testClass, testCase);
+
+        testCaseCollector.collectFromNamespaces(namespaces.toArray(new String[0]));
+        NamespaceUtils.deleteNamespacesFromSet(namespaces, testClass, testCase);
```

Comment:
Why does a class called TestLogCollector have a side effect?   Intuitively, you'd just expect it to snapshot state, not alter it.   The code violates least suprise.